### PR TITLE
Clean boost

### PR DESCRIPTION
**Pull request by bredelings**
A few small cleanups for the boost code.

### CODE REVIEW

File: mesonbuild/dependencies/misc.py [hunk 1]
```diff
@@ -147,22 +147,7 @@ def __init__(self, environment, kwargs):
...
-                             'This will be an error in the future')
-                remove.append(m)
-
-        self.requested_modules = [x for x in self.requested_modules if x not in remove]
-        invalid_modules = [x for x in invalid_modules if x not in remove]
-
-        if invalid_modules:
-            mlog.log(mlog.red('ERROR:'), 'Invalid Boost modules: ' + ', '.join(invalid_modules))
-            self.log_fail()
+        if check_invalid_modules(self):
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[Flake8]__

[F821] undefined name 'check_invalid_modules'

File: mesonbuild/dependencies/misc.py [hunk 2]
```diff
@@ -359,15 +364,19 @@ def detect_lib_modules_win(self):
                     fname = os.path.basename(entry)
                     self.lib_modules[self.modname_from_filename(fname)] = [fname]
 
+    def abi_tag(self):
+        if for_windows(self.want_cross, self.env):
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[Flake8]__

[F821] undefined name 'for_windows'

File: mesonbuild/dependencies/misc.py [hunk 3]
```diff
@@ -359,15 +364,19 @@ def detect_lib_modules_win(self):
...
     def detect_lib_modules_nix(self):
         all_found = True
         for module in self.requested_modules:
-            args = None
-            libname = 'boost_' + module
-            if self.is_multithreading and mesonlib.for_darwin(self.want_cross, self.env):
-                # - Linux leaves off -mt but libraries are multithreading-aware.
-                # - Mac requires -mt for multithreading, so should not fall back to non-mt libraries.
-                libname = libname + '-mt'
+            libname = 'boost_' + module + abi_tag(self)
```
Comment (ignatenkobrain) [inline mesonbuild/dependencies/misc.py]: __[Flake8]__

[F821] undefined name 'abi_tag'

File: mesonbuild/dependencies/misc.py [hunk 4]
```diff
@@ -359,15 +364,19 @@ def detect_lib_modules_win(self):
...
+
     def detect_lib_modules_nix(self):
         all_found = True
         for module in self.requested_modules:
-            args = None
-            libname = 'boost_' + module
-            if self.is_multithreading and mesonlib.for_darwin(self.want_cross, self.env):
-                # - Linux leaves off -mt but libraries are multithreading-aware.
-                # - Mac requires -mt for multithreading, so should not fall back to non-mt libraries.
-                libname = libname + '-mt'
```
Comment (jpakkane) [inline mesonbuild/dependencies/misc.py]: These comments are important. Please don't delete them but move them to the abi_tag function instead.

Comment (bredelings) [reply mesonbuild/dependencies/misc.py]: Sure, good point.  I put these back added a bit to the documentation earlier in the file.

**Pull request merged**